lib/custom_set_cases.rb

```diff
@@ -0,0 +1,89 @@
+class CustomSetCase < OpenStruct
+  def name
+    'test_%s' % description.gsub(/ |-/, '_')
+  end
+
+  def test_body
+    instance_eval section
```

Comment:
Could you use `send` instead of `instance_eval` ?
This makes it clearer that you're just calling one of the methods of `CustomSetCase` and you no longer have to worry too much if you're ever going to get something nasty in through `section`